easybuild/easyblocks/c/cp2k.py

```diff
@@ -394,32 +400,40 @@ def configure_common(self):
                     raise EasyBuildError("Building the libint wrapper failed")
                 libint_wrapper = '%s/libint_cpp_wrapper.o' % libinttools_path
 
-            # determine LibInt libraries based on major version number
-            libint_maj_ver = get_software_version('LibInt').split('.')[0]
+            # determine Libint libraries based on major version number
+            libint_maj_ver = get_software_version('Libint').split('.')[0]
             if libint_maj_ver == '1':
                 libint_libs = "$(LIBINTLIB)/libderiv.a $(LIBINTLIB)/libint.a $(LIBINTLIB)/libr12.a"
             elif libint_maj_ver == '2':
                 libint_libs = "$(LIBINTLIB)/libint2.a"
             else:
                 raise EasyBuildError("Don't know how to handle libint version %s", libint_maj_ver)
-            self.log.info("Using LibInt version %s" % (libint_maj_ver))
+            self.log.info("Using Libint version %s" % (libint_maj_ver))
 
             options['LIBINTLIB'] = '%s/lib' % libint
             options['LIBS'] += ' %s -lstdc++ %s' % (libint_libs, libint_wrapper)
 
         else:
-            # throw a warning, since CP2K without LibInt doesn't make much sense
-            self.log.warning("LibInt module not loaded, so building without LibInt support")
+            # throw a warning, since CP2K without Libint doesn't make much sense
+            self.log.warning("Libint module not loaded, so building without Libint support")
 
 
         libxc = get_software_root('libxc')
         if libxc:
             cur_libxc_version = get_software_version('libxc')
+            if LooseVersion(self.version) >= LooseVersion('6.1'):
+                LIBXC_MIN_VERSION = '4.0.3'
+                dflags = ' -D__LIBXC'
+            else:
+                dflags = ' -D__LIBXC2'
+
             if LooseVersion(cur_libxc_version) < LooseVersion(LIBXC_MIN_VERSION):
-                raise EasyBuildError("CP2K only works with libxc v%s (or later)", LIBXC_MIN_VERSION)
+                raise EasyBuildError("This version of CP2K is not compatible with libxc < %s" % LIBXC_MIN_VERSION)
 
-            options['DFLAGS'] += ' -D__LIBXC2'
-            if LooseVersion(cur_libxc_version) >= LooseVersion('2.2'):
+            options['DFLAGS'] += dflags
```

Comment:
@damianam This line should be moved up, above the version check for `libxc` (to keep things together better)